lib/client.js

```diff
@@ -1820,6 +1820,11 @@ function writeH2 (client, session, request) {
   })
 
   stream.on('data', (chunk) => {
+    // Aborting a request does not abort
+    // the stream, so we need to check and destroy it if request
+    // is aborted.
+    if (request.aborted) util.destroy(stream, new RequestAbortedError())
```

Comment:
I'm not super convinced with the solution but couldn't find other way. So I'm open to suggestions

Reply:
Why not register this logic in the closure sent in onConnect?

Reply:
Do you mean here https://github.com/nodejs/undici/blob/3a77cbb471b76175898d8b094beb9b40ceadd671/lib/client.js#L1684?

Reply:
I think it would be better to remove the listener instead

Reply:
>Do you mean here

Hmm, yeah it makes sense; maybe we can also call it just and only if the stream is ready or wait until the `ready` event triggers.

>I think it would be better to remove the listener instead

Will the data remain buffered there, or what will be the direct consequence of removing the listener? Just to understand the reason behind better 😅

Reply:
The current patch seems like we'd immediately call request.onData in the next line (since destroy only schedules an 'error' event), which would then hit the same `assert(!this.aborted)`, and experimentally this is what I run into